app/src/androidTest/java/ch/hikemate/app/ui/profile/ProfileScreenTest.kt

```diff
@@ -0,0 +1,150 @@
+
+    composeTestRule.setContent {
+      ProfileScreen(profileViewModel = profileViewModel, navigationActions = navigationActions)
+    }
+  }
+
+  @Test
+  fun isEverythingDisplayed() {
+    composeTestRule.onNodeWithTag(TEST_TAG_SIDEBAR_BUTTON).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_TITLE).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_NAME).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_EMAIL).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_JOIN_DATE).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_EDIT_PROFILE_BUTTON).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_SIGN_OUT_BUTTON).assertIsDisplayed()
+  }
+
+  @Test
+  fun checkCorrectProfileInfo() {
+    val profile =
+        Profile(
+            id = "1",
+            name = "John Doe",
+            email = "john.doe@gmail.com",
+            hikingLevel = HikingLevel.INTERMEDIATE,
+            joinedDate = Timestamp.now())
+    `when`(profileRepository.getProfileById(any(), any(), any())).thenAnswer {
+      val onSuccess = it.getArgument<(Profile) -> Unit>(1)
+      onSuccess(profile)
+    }
+    profileViewModel.getProfileById("1")
+
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_NAME).assertTextEquals(profile.name)
+    composeTestRule.onNodeWithTag(ProfileScreen.TEST_TAG_EMAIL).assertTextEquals(profile.email)
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL)
+        .assertTextContains(
+            profile.hikingLevel.toString().lowercase(), substring = true, ignoreCase = true)
+    val current = LocalDate.now()
+    val currentDateDayOfMonth = current.dayOfMonth
+    val currentDateMonthText = current.month.toString().lowercase()
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_JOIN_DATE)
+        .assertTextContains(currentDateDayOfMonth.toString(), substring = true, ignoreCase = true)
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_JOIN_DATE)
+        .assertTextContains(currentDateMonthText, substring = true, ignoreCase = true)
+  }
+
+  @Test
+  fun checkExpertHikingLevel() {
+    val profile =
+        Profile(
+            id = "1",
+            name = "John Doe",
+            email = "john.doe@gmail.com",
+            hikingLevel = HikingLevel.EXPERT,
+            joinedDate = Timestamp.now())
+    `when`(profileRepository.getProfileById(any(), any(), any())).thenAnswer {
+      val onSuccess = it.getArgument<(Profile) -> Unit>(1)
+      onSuccess(profile)
+    }
+
+    profileViewModel.getProfileById("1")
+
+    composeTestRule
+        .onNodeWithTag(ProfileScreen.TEST_TAG_HIKING_LEVEL)
+        .assertTextContains(
+            profile.hikingLevel.toString().lowercase(), substring = true, ignoreCase = true)
```

Comment:
Warning: In case of use of translation, this test will no longer hold. Have you any way to improve it?

Reply:
Done!